### PR TITLE
More fix for xss and sql injection

### PR DESCRIPTION
**Pull request by jcfr**

### CODE REVIEW

**Comment by jcfr** [comment]
@mgrauer When you have a chance, you could merge the style fix ? Thanks

**Comment by mgrauer** [comment]
@jcfr  Done. 

The way to do this is, click on the details of the failing styleci, login (using GH), click on create fix PR, click on view fix PR, this will take you to a new PR merging the style fix into your base branch which is the original PR, merge the style fix PR and delete the branch.

**Comment by jcfr** [comment]
Thanks Mike

>  login (using GH)

I missed that part and assumed i needed special credential but that makes sense.

**Comment by thewtex** [comment]
:+1: LGTM!

File: core/GlobalController.php [hunk 1]
```diff
@@ -283,4 +283,22 @@ public function getFormAsArray(Zend_Form $form)
 
         return $array;
     }
+
+  /**
+   * Return a sanitized request parameter useful to prevent XSS attack.
+   */
+  public function getSafeParam($paramName, $trim = false)
```
Comment (jamiesnape) [inline core/GlobalController.php]: Indent.

File: core/GlobalController.php [hunk 2]
```diff
@@ -283,4 +283,22 @@ public function getFormAsArray(Zend_Form $form)
...
+  {
+      $value = $this->getParam($paramName);
+      if ($trim) {
+          $value = trim($value);
+      }
+
+      return $this->sanitize($value);
+  }
+
+    protected function sanitize($value)
```
Comment (jamiesnape) [inline core/GlobalController.php]: Needs docblock.

File: core/controllers/UserController.php [hunk 3]
```diff
@@ -1253,8 +1261,8 @@ public function emailregisterAction()
         if ($this->_request->isPost()) {
             $this->disableLayout();
             $this->disableView();
-            $firstName = trim($this->getParam('firstName'));
-            $lastName = trim($this->getParam('lastName'));
+            $firstName = htmlentities(trim($this->getParam('firstName')));
+            $lastName = htmlentities(trim($this->getParam('lastName')));
```
Comment (jamiesnape) [inline core/controllers/UserController.php]: Use `getSafeParam(..., true)`.

**Pull request merged**